### PR TITLE
DMP-99 Create daily list endpoints

### PR DESCRIPTION
**Pull request by cmagowan-hmcts**
https://tools.hmcts.net/jira/browse/DMP-99

Add dailyList endpoints

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -236,5 +250,21 @@ rootProject.tasks.named("processSmokeTestResources") {
...
+  distributionType = Wrapper.DistributionType.ALL
+}
+
+
+openApiGenerate {
+  generatorName = "spring"
+  inputSpec = "$rootDir/src/main/resources/openapi/dailyList.yaml".toString()
+  outputDir = "$buildDir/generated/openapi".toString()
+  apiPackage = "uk.gov.hmcts.darts.dailylist.api"
+  //invokerPackage = "uk.gov.hmcts.darts.invoker"
```
Comment (davet1985) [inline build.gradle:262]: is this needed? or can it be removed?

File: config/checkstyle/checkstyle.xml [hunk 2]
```diff
@@ -26,7 +26,9 @@
   <module name="FileTabCharacter">
     <property name="eachLine" value="true"/>
   </module>
-
+  <module name="BeforeExecutionExclusionFileFilter">
+    <property name="fileNamePattern"  value=".*[\\/]generated[\\/].*$||.*[\\/]controllers[\\/].*$"/>
```
Comment (davet1985) [inline config/checkstyle/checkstyle.xml:30]: does this exclude files under "controllers" from being checking? is that something we want/need?

File: src/integrationTest/java/uk/gov/hmcts/darts/common/controller/GetWelcomeTest.java [hunk 3]
```diff
@@ -24,4 +27,115 @@ void welcomeRootEndpoint() throws Exception {
 
         assertThat(response.getResponse().getContentAsString()).startsWith("Welcome");
     }
+
+    @Test
+    void dailyListAddDailyListEndpoint() throws Exception {
+        MockHttpServletRequestBuilder requestBuilder = post("/dailylist/addDailyList")
+            .contentType(MediaType.APPLICATION_JSON_VALUE)
+            .queryParam("source_system", "CPP")
+            .content("{\n" +
```
Comment (davet1985) [inline src/integrationTest/java/uk/gov/hmcts/darts/common/controller/GetWelcomeTest.java:36]: Would be cleaner if you could move this content to a separate file somewhere?

File: config/checkstyle/checkstyle.xml (hunk 2 above)
Comment (cmagowan-hmcts) [reply config/checkstyle/checkstyle.xml:30]: yeah, the controllers have annotations etc in them that was generated from openapiGenerator, so it generates lines that are longer than the recommended 120 characters etc. i can't remember if that was the only rule broken, so either we remove the rule or exclude the file.

File: src/integrationTest/java/uk/gov/hmcts/darts/common/controller/GetWelcomeTest.java [hunk 4]
```diff
@@ -24,4 +27,115 @@ void welcomeRootEndpoint() throws Exception {
 
         assertThat(response.getResponse().getContentAsString()).startsWith("Welcome");
     }
+
+    @Test
+    void dailyListAddDailyListEndpoint() throws Exception {
```
Comment (davet1985) [inline src/integrationTest/java/uk/gov/hmcts/darts/common/controller/GetWelcomeTest.java:32]: I think this should probably be moved into a different file? Like `DailyListTest.java` or something?

**Review by davet1985 — COMMENTED** [review]
A few minor comments, looks good.

File: src/integrationTest/java/uk/gov/hmcts/darts/common/controller/GetWelcomeTest.java (hunk 3 above)
Comment (cmagowan-hmcts) [reply src/integrationTest/java/uk/gov/hmcts/darts/common/controller/GetWelcomeTest.java:36]: i tried that, and errors appeared in the build. Seems to be some sort of build config problem. thinks any file added is a duplicate.

File: config/checkstyle/checkstyle.xml (hunk 2 above)
Comment (davet1985) [reply config/checkstyle/checkstyle.xml:30]: We can just suppress the line length check for those files

```
  <module name="SuppressionSingleFilter">
    <property name="files" value=".*[\\/]controllers[\\/].*$" />
    <property name="checks" value="LineLength" />
  </module>
```

File: src/integrationTest/java/uk/gov/hmcts/darts/common/controller/GetWelcomeTest.java (hunk 4 above)
Comment (cmagowan-hmcts) [reply src/integrationTest/java/uk/gov/hmcts/darts/common/controller/GetWelcomeTest.java]: done

File: src/integrationTest/java/uk/gov/hmcts/darts/common/controller/GetWelcomeTest.java (hunk 3 above)
Comment (cmagowan-hmcts) [reply src/integrationTest/java/uk/gov/hmcts/darts/common/controller/GetWelcomeTest.java]: managed to find the setting that was needed to fix the error.

File: config/checkstyle/checkstyle.xml (hunk 2 above)
Comment (cmagowan-hmcts) [reply config/checkstyle/checkstyle.xml]: done

File: build.gradle (hunk 1 above)
Comment (cmagowan-hmcts) [reply build.gradle]: removed

**Comment by davet1985** [comment]
@cmagowan-hmcts this looks good now, could you bring it up to date with master branch?

**Review by davet1985 — APPROVED** [review]
LGTM

**Pull request merged**